profiles/ug/modules/ug/ug_system/ug_system.install

```diff
@@ -1,21 +1,163 @@
+    }
+
+    // ------------------------
+    // RECAPTCHA SECRET KEY
+    // ------------------------
+    $recaptcha_secret_key = variable_get('recaptcha_secret_key');
+    $requirements['ug_system_recaptcha_secret_key'] = array(
+      'title' => t('UG recaptcha Secret key'),
+      'description' => t('This ensures the recaptcha secret key is correct'),
+    );
+    if ($recaptcha_secret_key != '6LfhO_gSAAAAAKdSU9SUzOHwBjueKx-2vjlQR0L5')  {
+      $requirements['ug_system_recaptcha_secret_key']['value'] = t('Incorrect key');
+      $requirements['ug_system_recaptcha_secret_key']['severity'] = REQUIREMENT_ERROR;
+      $requirements['ug_system_recaptcha_secret_key']['description'] = t('recaptcha Secret key is missing/incorrect');
+    }
+
+    // ------------------------
+    // ERROR NOTIFICATIONS
+    // ------------------------
+    $error_level = variable_get('error_level');
+    $requirements['ug_system_error_level'] = array(
+      'title' => t('UG Error Messages'),
+      'description' => t('Checks if error messages is turned off'),
+    );
+    if ($error_level != '0'){
+      $requirements['ug_system_error_level']['value'] = t('Not configured');
+      $requirements['ug_system_error_level']['severity'] = REQUIREMENT_ERROR;
+      $requirements['ug_system_error_level']['description'] = t('Turn error messages off. Value should be 0(None)');
+    }
+
+    // ------------------------
+    // CACHE LIFETIME
+    // ------------------------
+    $cache_lifetime = variable_get('cache_lifetime');
+    $requirements['ug_system_cache_lifetime'] = array(
+      'title' => t('UG Minimum Cache lifetime'),
+      'description' => t('This ensures that the cache lifettime is enabled under Configuration / Performance and it is set to 5 mins'),
+    );
+    if ($cache_lifetime != '300') {
+      $requirements['ug_system_cache_lifetime']['value'] = t('Caching is not enabled');
+      $requirements['ug_system_cache_lifetime']['severity'] = REQUIREMENT_ERROR;
+      $requirements['ug_system_cache_lifetime']['description'] = t('Set to 5 min (300)');
+    }
+
+    // ------------------------
+    // CACHE MAXIMUM AGE
+    // ------------------------
+    $page_cache_maximum_age = variable_get('page_cache_maximum_age');
+    $requirements['ug_system_page_cache_maximum_age'] = array(
+      'title' => t('UG Expiration of cached pages'),
+      'description' => t('This ensures page caching is enabled under Configuration / Performance and it is set to 5 mins'),
+    );
+    if ($page_cache_maximum_age != '300') {
+      $requirements['ug_system_page_cache_maximum_age']['value'] = t('Caching is not enabled');
+      $requirements['ug_system_page_cache_maximum_age']['severity'] = REQUIREMENT_ERROR;
+      $requirements['ug_system_page_cache_maximum_age']['description'] = t('set to 5 min (300)');
+    }
+
+    // ------------------------
+    // LDAP CONFIGURATION
+    // ------------------------
+    $ldap = db_query("select * from {ldap_servers}");
```

Comment:
What if there is more than one row returned? Should this include a "where id = 'ug'" condition or similar?